index.go

```diff
@@ -0,0 +1,31 @@
+package imgutil
+
+import (
+	"github.com/google/go-containerregistry/pkg/name"
+)
+
+// ImageIndex an Interface with list of Methods required for creation and manipulation of v1.IndexManifest
+type ImageIndex interface {
+	// getters
+
+	Annotations(digest name.Digest) (annotations map[string]string, err error)
+	Architecture(digest name.Digest) (arch string, err error)
+	Features(digest name.Digest) (features []string, err error)
+	OS(digest name.Digest) (os string, err error)
+	OSFeatures(digest name.Digest) (osFeatures []string, err error)
+	OSVersion(digest name.Digest) (osVersion string, err error)
+	Variant(digest name.Digest) (osVariant string, err error)
+
+	// setters
+
+	SetAnnotations(digest name.Digest, annotations map[string]string) error
+
+	// misc
+
+	Add(repoName string, ops ...func(options *IndexAddOptions) error) error
```

Comment:
Shouldn't we do something like `Add(image v1.Image, ...) and make the caller responsible for instantiating the image? That would make this work across layout/remote etc.

Reply:
@jjbustamante WDYT?